src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java

```diff
@@ -126,6 +127,47 @@ private Optional<InterfaceTypeDefinition> getInterfaceTypeDefinition(TypeDefinit
         return Optional.empty();
     }
 
+    public List<String> getTypesImplementingInterface(@NotNull PsiElement psiElement, @NotNull String interfaceName) {
+        TypeDefinitionRegistry registry = getRegistry(psiElement);
+
+        // Check if the parentType is actually an interface
+        Optional<InterfaceTypeDefinition> interfaceType = getInterfaceTypeDefinition(registry, interfaceName);
+        if (!interfaceType.isPresent()) {
+            return new ArrayList<>(); // Not an interface
+        }
+
+        List<String> implementingTypes = new ArrayList<>();
+
+        // Search all object types to find those implementing this interface
+        registry.types().values().forEach(typeDefinition -> {
+            if (typeDefinition instanceof ObjectTypeDefinition) {
+                ObjectTypeDefinition objectType = (ObjectTypeDefinition) typeDefinition;
+                // Check if this type implements the interface
+                objectType.getImplements().stream()
+                    .filter(type -> type instanceof TypeName && ((TypeName) type).getName().equals(interfaceName))
+                    .findAny()
+                    .ifPresent(t -> implementingTypes.add(objectType.getName()));
```

Comment:
Minor but I think you can avoid an `Optional` creation and just use `forEach` here. Feel free to ignore.

TL;DR

Whether you use `findAny`, `findFirst` or `forEach`, it's not really clear to the reader whether "there can be only one" (Highlander!) or more than one, which is just the result of the `Stream` API.